ldms/src/ldmsd/ldmsd_request.c

```diff
@@ -766,6 +767,28 @@ static int string2attr_list(char *str, struct attr_value_list **__av_list,
 	return rc;
 }
 
+int validate_ldmsd_req(ldmsd_req_ctxt_t reqc, ldmsd_req_hdr_t rh)
+{
+	ldmsd_req_attr_t attr;
+	unsigned int off;
+	/* Validate length of request header and request. */
+	if (rh->rec_len <= sizeof(*rh))
+		return 0;
+	attr = (ldmsd_req_attr_t)(rh+1);
+	while (attr->discrim)
+	{
+		off = sizeof(*rh);
+		/* Check attr doesn't reference beyond end of request buffer */
+		if (off + attr->attr_len > rh->rec_len)
+			return 0;
+		attr = (ldmsd_req_attr_t)&attr->attr_value[attr->attr_len];
+	}
+	/* Check null termination of last attr string */
```

Comment:
This check is unnecessary because the while(attr->discrim) already checks this. Also '\0' is the character 0, not an unsigned long.